test/integration/test/deck_test.go

```diff
@@ -379,137 +381,137 @@ func TestRerun(t *testing.T) {
 		t.Fatalf("Failed creating clients for cluster %q: %v", clusterContext, err)
 	}
 
-	rerunJobConfig := fmt.Sprintf(rerunJobConfigTemplate, jobName, "foo")
-	if err := updateJobConfig(context.Background(), kubeClient, rerunJobConfigFile, []byte(rerunJobConfig)); err != nil {
-		t.Fatalf("Failed update job config: %v", err)
-	}
+	ctx := context.Background()
 
-	// Now we are waiting on Horologium to create the first prow job so that we
-	// can rerun from.
-	// Horologium itself is pretty good at handling the configmap update, but
-	// not kubelet, according to
-	// https://github.com/kubernetes/kubernetes/issues/30189 kubelet syncs
-	// configmap updates on existing pods every minute, which is a long wait.
-	// The proposed fix in the issue was updating the deployment, which imo
-	// should be better handled by just refreshing pods.
-	// So here comes forcing restart of horologium pods.
-	if err := refreshProwPods(kubeClient, context.Background(), "horologium"); err != nil {
-		t.Fatalf("Failed refreshing horologium pods: %v", err)
-	}
-	// Same with deck
-	if err := refreshProwPods(kubeClient, context.Background(), "deck"); err != nil {
-		t.Fatalf("Failed refreshing deck pods: %v", err)
+	redeployJobConfig := func(jobConfig string) {
+		if err := updateJobConfig(ctx, kubeClient, rerunJobConfigFile, []byte(jobConfig)); err != nil {
+			t.Fatalf("Failed update job config: %v", err)
+		}
+
+		// Now we are waiting on Horologium to create the first prow job so that we
+		// can rerun from.
+		// Horologium itself is pretty good at handling the configmap update, but
+		// not kubelet, according to
+		// https://github.com/kubernetes/kubernetes/issues/30189 kubelet syncs
```

Comment:
n.b. the linked issue says some semantically useless annotation update should kick the kubelet